src/main.py

```diff
@@ -474,22 +474,15 @@ def main(input_data: str, step_length: int, path_param: str, cores: int, solver=
             new_res_cap = mu.get_new_capacity_lifetime(op_life, new_cap)
             res_cap = mu.merge_res_capacites(old_res_cap, new_res_cap)
             
-            # overwrite residual capacity values for all subsequent steps
+            # overwrite residual capacity values for next step
             next_step = step + 1
-            while next_step < num_steps:
 
-                step_res_cap = res_cap.loc[res_cap["YEAR"].isin(actual_years_per_step[next_step])]
-                
-                # no more res capacity to pass on
-                if step_res_cap.empty:
-                    break
-                
+            step_res_cap = res_cap.loc[res_cap["YEAR"].isin(modelled_years_per_step[next_step])]
```

Comment:
Also, maybe please just check that I am passing the capacity over the correct time frame! I am passing it over the modelled years, rather than the actual years. If this is incorrect, please revert these two changes!